src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -803,29 +803,40 @@ def max_timestamp(aggregate_filter, org_id, use_case_id, alias=None):
 
 
 def on_demand_failure_rate_snql_factory(aggregate_filter, org_id, use_case_id, alias=None):
+    """Divide the number of transactions that failed from the total."""
     return Function(
         "divide",
         [
+            on_demand_failure_count_snql_factory(
+                aggregate_filter, org_id, use_case_id, "failure_count"
+            ),
```

Comment:
It's hard to see this in the diff but I've extracted these lines into `on_demand_failure_count_snql_factory`
https://github.com/getsentry/sentry/blob/f78966318f68efc951a1f09c4b48574302556534/src/sentry/snuba/metrics/fields/snql.py#L809-L827